app/controllers/course/leaderboards_controller.rb

```diff
@@ -3,12 +3,17 @@ class Course::LeaderboardsController < Course::ComponentController
   before_action :check_component
   before_action :load_settings
   before_action :add_leaderboard_breadcrumb
+  before_action :check_group_leaderboard_settings, only: [:groups]
 
   def show #:nodoc:
     preload_course_levels
     load_course_users
```

Comment:
@lowjoel Actually `load_course_users` should be part of the action, no? I think that is more consistent with the rest of the app and similar to the comment below about setting `@groups = @course.groups` as part of the action itself.